staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```
@@ -129,6 +116,19 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
 		userInfo, _ := request.UserFrom(ctx)
+
+		_, name, err := scope.Namer.Name(req)
+		if err != nil {
+			if err.Error() != errEmptyName.Error() {
+				scope.err(err, w, req)
+				return
+			}
+
+			// On create, get name from new object if unset
+			if len(name) == 0 {
+				_, name, err = scope.Namer.ObjectName(obj)
```

if this errors, the error is ignored